src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -758,9 +758,18 @@ static int rbd_nbd(int argc, const char *argv[])
 
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
-			 CODE_ENVIRONMENT_DAEMON,
-			 CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
+
+  code_environment_t code_env = CODE_ENVIRONMENT_UTILITY;
+  int env_flag = 0;
+  for (auto arg : args) {
+    if (strcmp(arg, "map") == 0) {
```

Comment:
Theoretically, "map" can be an image name, or a cluster name, when it is specified as `--cluster map`. Right now this is still only useful for map command, so the program should behave properly, still it is not very nice.

Actually I don't mind if Jason agrees, but what if we do this way instead:

- Don't call global_init here, and filter out common ceph options from args using md_config_t::parse_argv():
```
md_config_t().parse_argv(args);
```
- Put argc and argv further to do_map(), and call global_init() there.

So global context initialization (and socket creation) would be only for "map" command.